src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs

```diff
@@ -0,0 +1,95 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.OperationalInsights
+{
+    [Cmdlet(VerbsData.Restore, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "OperationalInsightsWorkspace", SupportsShouldProcess = true), OutputType(typeof(PSWorkspace))]
+    public class RestoreAzureOperationalInsightsWorkspaceCommand : OperationalInsightsBaseCmdlet
+    {
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The workspace name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The geographic region that the workspace will be created in.")]
+        [LocationCompleter("Microsoft.OperationalInsights/workspaces")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The service tier of the workspace.")]
+        [PSArgumentCompleter("free", "standard", "premium", "pernode", "standalone", "pergb2018")]
+        public string Sku { get; set; }
```

Comment:
Why did you add all those parameters to the Restore command? This should be much simple, just get the name, group and location. All the rest can be updated latter after the workspace has been restored.

Reply:
Followed description from issue:
https://github.com/Azure/azure-cli/issues/13456

"Calling PUT /workspaces to re-create the workspace. If you specify just the location and empty properties, the workspace is recovered with the same properties it had when it was deleted. You can optionally provide properties that will override the ones we keep for that workspace. The behavior is PATCH like."

Reply:
Behavior is a little strange that SKU can be changed during restore. I agree to Yoram.